GLSP-1266 Fix handling of zero values in builder functions

ndoschek opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/eclipse-glsp/glsp/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See [SECURITY.md](https://github.com/eclipse-glsp/glsp/blob/master/SECURITY.md),
to learn how to report vulnerabilities.
-->

#### What it does

<!-- Include relevant issues and describe how they are addressed. -->

- Fix handling of zero values in `GBoundsAwareBuilder` functions
- Fix handling of zero values in `GAlignableBuilder` function
- Adds unit tests for those updated functions

Resolves eclipse-glsp/glsp/issues/1266

#### Changelog

<!-- Please check, when if it applies to your change. -->

- [x] This PR should be mentioned in the changelog
- [ ] This PR introduces a breaking change (if yes, provide more details below for the changelog and the migration guide)

tortmayr reviewed (changes requested):
Looks good to me. Could you please also make the same adjustment for the `GAlignableBuilder` functions?
Also please update the `Resolves` comment in your commit to `Resolves  eclipse-glsp/glsp/issues/1266` otherwise autoclosing on merge won't work.

ndoschek commented:
Sure, thanks for the review @tortmayr!

ndoschek commented:
I found some motivation and added a few unit tests for the functions I updated, would be great if you could have a quick look on those as well when you get a chance. Thanks!

tortmayr reviewed (approved):
Thanks. More test coverage is always appreciated 😉
